sparkmagic/sparkmagic/livyclientlib/sqlquery.py

```diff
@@ -49,13 +51,17 @@ def execute(self, session):
         self._spark_events.emit_sql_execution_start_event(session.guid, session.kind, session.id, self.guid,
                                                           self.samplemethod, self.maxrows, self.samplefraction)
         command_guid = ''
+        result = None
         try:
             command = self.to_command(session.kind, session.sql_context_variable_name)
             command_guid = command.guid
             (success, records_text, mimetype) = command.execute(session)
             if not success:
-                raise BadUserDataException(records_text)
-            result = records_to_dataframe(records_text, session.kind, self._coerce)
+                if conf.spark_statement_errors_are_fatal():
+                    raise SparkStatementException(records_text)
+                self.ipython_display.send_error(records_text)
```

Comment:
If `conf.spark_statement_errors_are_fatal() == False` and a `SparkStatementException` is raised then wouldn't this be handled and displayed by `handle_expected_exceptions`?